From dsd@ntlworld.com Mon May  8 17:45:23 2006
From: Daniel Drake <dsd@gentoo.org>
To: greg@kroah.com
Cc: <mdharm-usb@one-eyed-alien.net>
Subject: USB: usb-storage alauda: Fix transport info mismerge
Message-Id: <20060509004527.C0A5788A90E@zog.reactivated.net>
Date: Tue,  9 May 2006 01:45:27 +0100 (BST)

Unfortunately it looks like the transport entry for this subdriver was merged
into the protocol section, making this driver unusable :(

Signed-off-by: Daniel Drake <dsd@gentoo.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/usb.c |   18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

--- gregkh-2.6.orig/drivers/usb/storage/usb.c
+++ gregkh-2.6/drivers/usb/storage/usb.c
@@ -593,6 +593,15 @@ static int get_transport(struct us_data 
 		break;
 #endif
 
+#ifdef CONFIG_USB_STORAGE_ALAUDA
+	case US_PR_ALAUDA:
+		us->transport_name  = "Alauda Control/Bulk";
+		us->transport = alauda_transport;
+		us->transport_reset = usb_stor_Bulk_reset;
+		us->max_lun = 1;
+		break;
+#endif
+
 	default:
 		return -EIO;
 	}
@@ -648,15 +657,6 @@ static int get_protocol(struct us_data *
 		break;
 #endif
 
-#ifdef CONFIG_USB_STORAGE_ALAUDA
-	case US_PR_ALAUDA:
-		us->transport_name  = "Alauda Control/Bulk";
-		us->transport = alauda_transport;
-		us->transport_reset = usb_stor_Bulk_reset;
-		us->max_lun = 1;
-		break;
-#endif
-
 	default:
 		return -EIO;
 	}
